### ISSUE TITLE
Add the ability to support cross target dependencies

### ISSUE THREAD

**Issue by jerrymarino** [yonaskolb/XcodeGen#124]
I'd like to have multiple Xcode projects for a mammoth iOS app. Say, `MyApp.xcodeproj` `ThirdParty.xcodeproj` `Core.xcodeproj`.

Currently, there is dependencies between the targets in these projects. Say, `MyApp` depends on `ThirdParty` and `Core`. `Core` depends on `ThirdParty` as well.

I'd like for XcodeGen to be able to partition dependencies across multiple targets within the same workspace. Is this something reasonable or in the cards for XcodeGen.

**Comment by yonaskolb** [comment]
Yeah, that would be useful. It would require some sort of workspace spec. It’s not in the immediate future as it’s quite a bit of work and I’d like to get XcodeGen to a good and stable place first, but it’s definitely something for the future

**Comment by yonaskolb** [comment]
This should work using `implicit` `framework` dependencies as described here #224

**Comment by yonaskolb** [comment]
@rasterize how is your target linked? As in what sort of settings does Xcode apply?

**Comment by rastersize** [comment]
@yonaskolb Sorry for late reply.

My targets aren’t linked. I depend on a pre-build target that executes a few scripts. For example it updates the app’s `Info.plist` with the current version and build number.

Currently we have the main “App” Xcode project include the Pre-build project. The App project’s main target then has a dependency on the target in the Pre-build project.

We need to do that as part of a separate target. Otherwise the build will fail with Xcode’s new build system. Unless you know of some other method for doing so?

**Comment by toshi0383** [comment]
@rastersize 
> I depend on a pre-build target that executes a few scripts. For example it updates the app’s Info.plist with the current version and build number.
> We need to do that as part of a separate target. Otherwise the build will fail with Xcode’s new build system. Unless you know of some other method for doing so?

For Info.plists, how about using preprocessed Info.plist? You can modify Info.plist safely in Build Phase before or after compiling.
You can access preprocessed Info.plist at `$TEMP_DIR/Preprocessed-Info.plist`.
